src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/StratisHeaderVersionRule.cs

```diff
@@ -10,17 +14,27 @@ namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
     /// Checks if <see cref="StratisMain"/> network block's header has a valid block version.
     /// </summary>
     [HeaderValidationRule(CanSkipValidation = true)]
-    public class StratisHeaderVersionRule : ConsensusRule
+    public class StratisHeaderVersionRule : HeaderVersionRule
     {
         /// <inheritdoc />
-        /// <exception cref="ConsensusErrors.BadVersion">Thrown if block's version is outdated.</exception>
+        /// <exception cref="ConsensusErrors.BadVersion">Thrown if block's version is outdated or otherwise invalid.</exception>
         public override Task RunAsync(RuleContext context)
         {
             Guard.NotNull(context.ConsensusTip, nameof(context.ConsensusTip));
 
             ChainedHeader chainedHeader = context.ValidationContext.ChainedHeader;
 
             if (chainedHeader.Header.Version < 7)
+            {
+                this.Logger.LogTrace("(-)[BAD_VERSION_LOW]");
+                ConsensusErrors.BadVersion.Throw();
+            }
+
+            // A version of precisely 7 is what is currently generated by the legacy C++ nodes.
+
+            // Whereas BIP9 mandates that the top bits of version be 001. So a standard node should never generate
+            // block versions between 7 and 0x20000000.
+            if ((chainedHeader.Header.Version > 7) && (chainedHeader.Header.Version < ThresholdConditionCache.VersionbitsTopBits))
```

Comment:
Careful with this one, is this limit exactly like this on stratisX?

Reply:
@zeptin perhaps its a good idea to put a link to stratisX where version is checked.

Reply:
I don't think this is correct as X does not have deployment implemented.